include/simdjson/dom/element.h

```diff
@@ -206,6 +206,63 @@ class element {
   template<typename T>
   really_inline simdjson_result<T> get() const noexcept;
 
+  /**
+   * Get the value as the provided type (T).
+   *
+   * Supported types:
+   * - Boolean: bool
+   * - Number: double, uint64_t, int64_t
+   * - String: std::string_view, const char *
+   * - Array: dom::array
+   * - Object: dom::object
+   *
+   * @tparam T bool, double, uint64_t, int64_t, std::string_view, const char *, dom::array, dom::object
+   *
+   * @param value The variable to set to the value. May not be set if there is an error.
+   * @param error The variable to set to the error. Set to SUCCESS if there is no error.
+   *
+   * @returns true if the value was set, or false if there wsa an error.
+   */
+  template<typename T>
+  inline bool get(T &value, error_code &error) const noexcept;
```

Comment:
The returned value is not WARN_UNUSED which I think is good.

Reply:
On purpose :) Since we're not doing the redundant bool thing anymore, the new get() is `WARN_UNUSED error_code get(T&)`.

Reply:
i.e. I changed it after your suggestions :)